app/code/Magento/ConfigurableProduct/view/frontend/web/js/configurable.js

```diff
@@ -117,7 +117,7 @@ define([
         _overrideDefaults: function () {
             var hashIndex = window.location.href.indexOf('#');
 
-            if (hashIndex !== -1) {
+            if (hashIndex !== -1  && window.location.hash !== '') {
```

Comment:
Make sense to use `_.isEmpty` here as well.

Reply:
@omiroshnichenko 
Updated